include/NetAPI.h

```diff
@@ -288,6 +325,23 @@ struct ConnectionCapability
 	char hostname[];
 };
 
+/**
+ * Bind capability contents. Instances of this sealed with the NetworkBindKey
+ * sealing capability authorise binding to a specific server port.
+ */
+struct BindCapability
+{
+	/**
+	 * Allow to bind on an IPv6 or IPv4 interface.
+	 */
+	bool isIPv6;
```

Comment:
Maybe a pair of `PF_*` identifier and protocol index rather than just "{v4, v6}"?

Reply:
Do you mean a `uint8_t domain` field with `AF_INET` and `AF_INET6` equivalents? Why not. That being said it complicates the API and I'm not sure that any other potential value for this field would make sense

Reply:
I mean something like `struct BindCapability { uint8_t protocol_family; uint8_t protocol; uint16_t port }` so that I can capture at least `{IPv4, IPv6} x {TCP, UDP}`.  (We don't have to reuse `PF_*`/`AF_*` from Berkeley and we don't have to reuse their protocol namespaces within those families, but we could if convenient.)

Reply:
We treat TCP and UDP differently here (bind + accept doesn’t make sense for UDP).

I think I’d like a single capability to authorise v4 and v6, but what I really want probably isn’t possible: a single handle for a listening socket that lets you accept connections on either v4 or v6 if you’re supporting both.

The v6 case is more complex than v4 because you always have multiple v6 interfaces (ant least an link-local and routable one) and need to choose the one to listen on, so we might also need to expose that.

Reply:
I think that we concluded that, for starters, we'd assume a single interface and evolve bind capabilities as needs arise. Shall we stay with how things are for now?

Reply:
Probably fine for now, but maybe add a comment that this API is less stable than the rest of the stack and will change?

We also need corresponding rego functions to decode these capabilities.